spec/cached_resource/caching_spec.rb

```diff
@@ -59,22 +60,27 @@ class NotTheThing < ActiveResource::Base
         mock.get "/things/1.json?foo=bar", {}, @thing_json
         mock.get "/things/fded.json", {}, @string_thing_json
         mock.get "/things.json?name=42", {}, @nil_thing, 404
+        mock.get "/things.json?name=43", {}, @empty_array_thing
         mock.get "/things/4.json", {}, @date_thing_json
         mock.get "/not_the_things/1.json", {}, @not_the_thing_json
       end
     end
 
     it "should cache a response" do
       result = Thing.find(1)
-
       read_from_cache("thing/1").should == result
     end
 
-    it "should cache a nil response" do
+    it "shouldn't cache nil response" do
```

Comment:
Maybe add a second request and test that the subsequent request gets cached?

Reply:
Yes, to probe that we are caching not `nil` values, as expected. I will do. Thanks!

Reply:
I meant with same request params. That way we see that the following valid response gets cached.

Reply:
Oh, I understand now. You say mocking two different answers for the same request. First, return `nil` or `[]` and see that we are not catching, and then return a valid answer and check that we are catching. I don't think that will be needed, because we are checking the right answer in other test cases. Also, I couldn't find a way to mock two different answers to the same request params.